### PR TITLE
feat: current search provider

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- [x] current search provider icon
- [x] current search provider label
- [x] also add hover and tab functionality to toggle between providers

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

AS-34 #done

### CODE REVIEW

File: packages/shared/src/components/KeyboardShortcutLabel.tsx [hunk 1]
```diff
@@ -10,6 +10,19 @@ export const KeyboadShortcutLabel = ({
   className,
   keys,
 }: KeyboadShortcutLabelProps): ReactElement => {
+  const [mounted, setMounted] = useState(false);
+
+  useEffect(() => {
+    // keyboard shortcut buttons are most of the times dependant
+    // on browser features so we don't render them in SSR to
+    // avoid hydration mismatches
```
Comment (rebelchris) [inline packages/shared/src/components/KeyboardShortcutLabel.tsx:18]: Not for now, but should we maybe add a backlog ticket to make this into it's own hook for clarity down the line?

File: packages/shared/src/components/MainFeedLayout.tsx [hunk 2]
```diff
@@ -247,11 +254,27 @@ export default function MainFeedLayout({
...
     <FeedPageComponent
       className={classNames('relative', shouldUseFeedLayoutV1 && '!pt-0')}
     >
-      <div id={searchPanelGradientElementId} />
+      {isSearchPanelGradientActive && (
+        <FeedGradientBg
+          className={classNames(
+            'right-0 -z-1 hidden translate-x-0 opacity-0 transition-opacity duration-500 laptop:flex',
+            sidebarExpanded
+              ? '!-left-32 !max-w-[calc(100%)]'
```
Comment (rebelchris) [inline packages/shared/src/components/MainFeedLayout.tsx:272]: Why do you need the calc 100% here?

File: packages/shared/src/components/search/SearchPanel/SearchPanelAction.tsx [hunk 3]
```diff
@@ -16,8 +18,16 @@ const iconToProviderMap: Record<
   FunctionComponent<IconProps>
 > = {
   [SearchProviderEnum.Posts]: SearchIcon,
-  // TODO AS-3-search-merge replace with correct icon
-  [SearchProviderEnum.Chat]: AiIcon,
+  [SearchProviderEnum.Chat]: ({ className, ...rest }: IconProps) => (
+    <MagicIcon
+      className={classNames(
+        className,
+        'rounded-6 bg-gradient-to-t from-onion-40 to-cabbage-40 p-0.5 text-white',
```
Comment (rebelchris) [inline packages/shared/src/components/search/SearchPanel/SearchPanelAction.tsx:25]: Same comment from other PR on the colors

**Review by rebelchris — CHANGES_REQUESTED** [review]
Minor queries so far :)

File: packages/shared/src/components/KeyboardShortcutLabel.tsx (hunk 1 above)
Comment (capJavert) [reply packages/shared/src/components/KeyboardShortcutLabel.tsx:18]: Feel like hook would hide it and make it more magical and then in hindsight used whenever someone encountered any SSR issue vs most of the time you can solve it by other mechanisms. Agree we should keep it in check this is why I added comment for now to explain.

File: packages/shared/src/components/MainFeedLayout.tsx (hunk 2 above)
Comment (capJavert) [reply packages/shared/src/components/MainFeedLayout.tsx:272]: Good catch, I think I was playing with values and then left it.

File: packages/shared/src/components/KeyboardShortcutLabel.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/KeyboardShortcutLabel.tsx:18]: Yeah comment is perfect for now :)

**Review by rebelchris — APPROVED** [review]
Looks good to me

**Pull request merged**